src/snowflake/snowpark/_functions/scalar_functions.py

```diff
@@ -1495,3 +1496,280 @@ def h3_int_to_string(cell_id: ColumnOrName, _emit_ast: bool = True) -> Column:
+        target_resolution (ColumnOrName): The H3 resolution level (0-15) for the coverage.
+
+    Returns:
+        Column: An array of H3 cell IDs covering the geography or None if the operation cannot be performed.
+
+    Example::
+        >>> from snowflake.snowpark.functions import to_geography, lit
+        >>> df = session.create_dataframe([
+        ...     ("POLYGON((-122.4194 37.7749, -122.4094 37.7749, -122.4094 37.7849, -122.4194 37.7849, -122.4194 37.7749))",)
+        ... ], schema=["geog"])
+        >>> df.select(h3_try_coverage(to_geography(df["geog"]), lit(9)).alias("coverage")).collect()
+        [Row(COVERAGE='[\\n  617700169957507071,\\n  617700169958031359,\\n  617700169958293503,\\n  617700169961701375,\\n  617700169961963519,\\n  617700169962487807,\\n  617700169963012095,\\n  617700169963798527,\\n  617700169964060671,\\n  617700169964322815,\\n  617700169964584959,\\n  617700169964847103,\\n  617700169965109247,\\n  617700169965371391,\\n  617700170001809407,\\n  617700170002857983,\\n  617700170017800191\\n]')]
+    """
+    geography_col = _to_col_if_str(geography_expression, "h3_try_coverage")
+    resolution_col = _to_col_if_str(target_resolution, "h3_try_coverage")
+    return builtin("h3_try_coverage", _emit_ast=_emit_ast)(
+        geography_col, resolution_col
+    )
+
+
+@publicapi
+def h3_try_coverage_strings(
+    geography_expression: ColumnOrName,
+    target_resolution: ColumnOrName,
+    _emit_ast: bool = True,
+) -> Column:
+    """
+    Returns an array of H3 cell IDs as strings that cover the given geography at the specified resolution.
+    This function attempts to provide coverage of the geography using H3 cells, returning the cell IDs as strings.
+
+    Args:
+        geography_expression (ColumnOrName): A geography expression to be covered by H3 cells
+        target_resolution (ColumnOrName): The H3 resolution level for the coverage, ranging from 0 to 15
+
+    Returns:
+        Column: An array of H3 cell ID strings that cover the input geography
+
+    Examples::
+
+        >>> from snowflake.snowpark.functions import to_geography
+        >>> df = session.create_dataframe([
+        ...     "POLYGON((-122.4194 37.7749, -122.4094 37.7749, -122.4094 37.7849, -122.4194 37.7849, -122.4194 37.7749))"
+        ... ], schema=["geog"])
+        >>> df.select(h3_try_coverage_strings(to_geography(df["geog"]), lit(9)).alias("coverage")).collect()
+        [Row(COVERAGE='[\\n  "89283082803ffff",\\n  "8928308280bffff",\\n  "8928308280fffff",\\n  "89283082843ffff",\\n  "89283082847ffff",\\n  "8928308284fffff",\\n  "89283082857ffff",\\n  "89283082863ffff",\\n  "89283082867ffff",\\n  "8928308286bffff",\\n  "8928308286fffff",\\n  "89283082873ffff",\\n  "89283082877ffff",\\n  "8928308287bffff",\\n  "89283082aa7ffff",\\n  "89283082ab7ffff",\\n  "89283082b9bffff"\\n]')]
+    """
+    g = _to_col_if_str(geography_expression, "h3_try_coverage_strings")
+    r = _to_lit_if_not_column(target_resolution)
```

Comment:
I'm curious why do we need this function for what type of input?
 
I think in snowpark generally it's user's responsibility to pass a valid input -- users have the make the lit() call by themselves. this looks like a new pattern we're trying to introduce